File: 0202-happy-number/0202-happy-number.cpp
class Solution
{
public:
    bool isHappy(int n)
    {
        unordered_map<int,bool>mp(1e6);
       
       while(!mp[n]) {
            mp[n]=1;
            int newNum = 0;
            while (n > 0)
            {
                newNum += (n % 10) *(n%10);
                n = n / 10;
            }
            n = newNum;
        }
        return n==1;
    }
};

File: 0202-happy-number/0202-happy-number.java
class Solution {     
    public boolean isHappy(int n) {         
        HashSet<Integer> hashSet = new HashSet<>();         
        while (n != 1) {             
            n = getNextNum(n);             
            if (hashSet.contains(n)) return false;             
            hashSet.add(n);         
        }          
        return true;  // Reached 1 = happy number
    }      

    int getNextNum(int n) {         
        int nextNum = 0;         
        while (n > 0) {             
            int num = n % 10;      // Get last digit
            n /= 10;               // Remove last digit
            nextNum += num * num;  // Add square
        }         
        return nextNum;     
    } 
}